grblas/matrix.py

```diff
@@ -202,109 +251,130 @@ def new_from_values(cls, rows, columns, values, *, nrows=None, ncols=None, dup_o
     # to __setitem__ to trigger a call to GraphBLAS
     #########################################################
 
-    def ewise_add(self, other, op=NULL):
+    def ewise_add(self, other, op=None, *, require_monoid=True):
         """
         GrB_eWiseAdd_Matrix
 
         Result will contain the union of indices from both Matrices
+        Default op is binary.plus
+        Unless explicitly disabled, this method requires a monoid (directly or from a semiring).
+            The reason for this is that binary operators can create very confusing behavior when only
+            one of the two elements is present.
+            Examples: binary.minus where left=Missing and right=4 yields 4 rather than -4 as might be expected
+                      binary.gt where left=Missing and right=4 yields True
+                      binary.gt where left=Missing and right=0 yields False
+            The behavior is caused by grabbing the non-empty value and using it directly without performing
+            any operation. In the case of `gt`, the non-empty value is cast to a boolean.
+            For these reasons, users are required to be explicit when choosing this surprising behavior.
         """
         if not isinstance(other, Matrix):
             raise TypeError(f'Expected Matrix, found {type(other)}')
-        if op is NULL:
+        if op is None:
             op = binary.plus
```

Comment:
Should this be `monoid.plus`?  Will `binary.plus` work if `require_monoid=True`?

Reply:
Yes it should. I fixed it for Vector, but forgot for Matrix. Good eye!